backend/api/clients/delete.py

```diff
@@ -0,0 +1,41 @@
+from django.contrib import messages
+from django.shortcuts import render
+from django.contrib.auth.decorators import login_required
+from django.views.decorators.http import require_http_methods
+
+from backend.models import Client
+from backend.types.htmx import HtmxHttpRequest
+
+
+@require_http_methods(["DELETE"])
+@login_required
+def client_delete(request: HtmxHttpRequest, id: int):
+    try:
+        client = Client.objects.get(id=id)
```

Comment:
I've just realised there's no validation here. Could we please check if the client belongs to the user/team (if logged in as  team)

Reply:
> What about that HX-Reswap and HX-Retarget you have some more details on that ?

I spent about 2 hours today trying to get my proposed solution working but turns out HTMX doesn't abide by my "HX-Reswap" or "Re target" response headers so I'm not too sure how to get this working. Seems like the only way would be to do what we're doing - return 200 and just provide an error message.

I did look into the [response-targets Extension](https://htmx.org/extensions/response-targets/) but it was breaking random parts of the app and breaking OOB swaps for no reason, so it seems  a little buggy. 

So I guess we'll just use 200 from now on

Reply:
> I've just realised there's no validation here. Could we please check if the client belongs to the user/team (if logged in as team)

Generally if user by some strange accident (like 2 tabs where on one is still open dashboard with clients but this user have been logged out on 2nd tab and new one logged in and try to delete client from on 1st tab) would like to delete a client that doesn't belong to him then he will get 403 from django middleware since CSRF token wont match so basically our code won't even run to be able to check for that and user will go into a loop until he deny. If we would like to change that we would need to probably handle that through django CSRF_FAILURE_VIEW. I also can't think of the way how client that doesn't belong to a given user could be displayed in his dashboard other what I described.

Reply:
It's not really a good idea to rely on front end validation though - a user can just directly call the backend endpoint with someone else's and successfully call the endpoint. I'd rather do validation on the backend.

Reply:
Validation for user and team (if logged as team) have been added.